lib/modules/manager/asdf/upgradeable-tooling.ts

```diff
@@ -0,0 +1,271 @@
+import { AdoptiumJavaDatasource } from '../../datasource/adoptium-java';
+import { DockerDatasource } from '../../datasource/docker';
+import { GithubReleasesDatasource } from '../../datasource/github-releases';
+import { GithubTagsDatasource } from '../../datasource/github-tags';
+import { NodeDatasource } from '../../datasource/node';
+import * as nodeVersioning from '../../versioning/node';
+import * as regexVersioning from '../../versioning/regex';
+import * as semverVersioning from '../../versioning/semver';
+import type { PackageDependency } from '../types';
+
+type StaticTooling = Pick<
+  PackageDependency,
+  'depName' | 'datasource' | 'packageName' | 'versioning' | 'extractVersion'
+>;
+
+type DynamicTooling = (version: string) => StaticTooling | undefined;
+
+export const upgradeableTooling: Record<
+  string,
+  StaticTooling | DynamicTooling
+> = {
+  awscli: {
+    datasource: GithubTagsDatasource.id,
+    packageName: 'aws/aws-cli',
+    versioning: semverVersioning.id,
+  },
+  bun: {
+    datasource: GithubReleasesDatasource.id,
+    packageName: 'oven-sh/bun',
+    versioning: semverVersioning.id,
+    extractVersion: '^bun-v(?<version>\\S+)',
+  },
+  'cargo-make': {
+    datasource: GithubReleasesDatasource.id,
+    packageName: 'sagiegurari/cargo-make',
+    versioning: semverVersioning.id,
+  },
+  clojure: {
+    datasource: GithubTagsDatasource.id,
+    packageName: 'clojure/brew-install',
+    versioning: `${regexVersioning.id}:^(?<major>\\d+?)\\.(?<minor>\\d+?)\\.(?<patch>\\d+)\\.(?<build>\\d+)$`,
+  },
+  crystal: {
+    datasource: GithubTagsDatasource.id,
+    packageName: 'crystal-lang/crystal',
+    versioning: semverVersioning.id,
+  },
+  deno: {
+    datasource: GithubReleasesDatasource.id,
+    packageName: 'denoland/deno',
+    versioning: semverVersioning.id,
+    extractVersion: '^v(?<version>\\S+)',
+  },
+  direnv: {
+    datasource: GithubReleasesDatasource.id,
+    packageName: 'direnv/direnv',
+    versioning: semverVersioning.id,
+    extractVersion: '^v(?<version>\\S+)',
+  },
+  dprint: {
+    datasource: GithubReleasesDatasource.id,
+    packageName: 'dprint/dprint',
+    versioning: semverVersioning.id,
+  },
+  elixir: {
+    datasource: GithubTagsDatasource.id,
+    packageName: 'elixir-lang/elixir',
+    versioning: semverVersioning.id,
+    extractVersion: '^v(?<version>\\S+)',
+  },
+  elm: {
+    datasource: GithubReleasesDatasource.id,
+    packageName: 'elm/compiler',
+    versioning: semverVersioning.id,
+  },
+  erlang: {
+    datasource: GithubTagsDatasource.id,
+    packageName: 'erlang/otp',
+    extractVersion: '^OTP-(?<version>\\S+)',
+    versioning: `${regexVersioning.id}:^(?<major>\\d+?)\\.(?<minor>\\d+?)(\\.(?<patch>\\d+))?$`,
+  },
+  gauche: {
+    datasource: DockerDatasource.id,
```

Comment:
Should be a GitHub Release, no? https://github.com/sakuro/asdf-gauche/blob/71f574d7c934bc4977eec4fb8005e2ecd0d7ed0c/bin/install#L32

Reply:
Not sure: https://github.com/kachick/renovate-config-asdf/blob/main/plugins/gauche.json5#L11

Reply:
Hum, alright :sweat_smile: :+1: